backend/Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Build the manager binary
+FROM golang:1.16 as builder
+
+WORKDIR /workspace
+
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+
+# Download the go module artificats in the WORKDIR (similar to go mod download working locally)
+RUN go mod download
+
+# Copy the go source
+COPY *.go ./
+COPY config/ config/
+COPY routes/ routes/
+COPY util/ util/
+
+# To compile/build the go applications
+RUN make build
+RUN ./dist/managed-gitops-backend
```

Comment:
Wondering why we are executing the binary here?
Could we do a multi-stage Dockerfile build where..
* In stage 1, we build the binary and,
* In stage 2, we pack that binary into a lean public image ( like registry.access.redhat.com/ubi8/ubi-minimal ) for deployment.